app/src/main/java/com/example/kotlinmigration/viewmodels/MainViewModel.kt

```diff
@@ -64,4 +72,97 @@ class MainViewModel: ViewModel() {
             })
         }
     }
-}
\ No newline at end of file
+
+    fun insertDataToDatabase() {
+
+        viewModelScope.launch(Dispatchers.IO) {
+
+
+            _retrofitState.collect() {
+
+                when (it) {
+
+                    is RetrofitEvent.Successful -> {
+                        if (it.response != null) {
+                            for (item in it.response) {
```

Comment:
so I love what you have here inside of your `if(it.response != null)`

but this function is doing a little bit too much

I would recommend removing `_retrofitState.collect(){}` all together, and have this function's sole purpose be to insert data into the database

so you could add an argument to this function: `fun insertDataToDatabase(data: List<PostJsonItem>)`

and then just have the code inside of `if(it.response != null)`

Reply:
A lot more elegant, ur right it was overly complicated for no reason. Implemented